RELEASE.md

```diff
@@ -1,65 +1,51 @@
 # Upcoming Release 1.0.0
 ## Major features and improvements
-## Bug fixes and other changes
-## Breaking changes to the API
-## Upcoming deprecations for Kedro 1.0.0
-## Documentation changes
-## Community contributions
-
-# Release 1.0.0rc3
-
-## Major features and improvements
-Changed `DataCatalog.__getitem__` to raise `DatasetNotFoundError` for missing datasets, aligning with expected dictionary behavior.
```

Comment:
@SajidAlamQB I didn't include this because it was a bug fix in a new API between RC releases so it shouldn't be mentioned